arangod/Transaction/Manager.cpp

```diff
@@ -1020,6 +1018,7 @@ Result Manager::statusChangeWithTimeout(TransactionId tid,
 
 Result Manager::commitManagedTrx(TransactionId tid,
                                  std::string const& database) {
+  READ_LOCKER(guard, _hotbackupCommitLock);
```

Comment:
Just wondering if other places than `transaction::Manager::commitManagedTrx(...)` should also prevent transactions from committing. At least this is not the only place that will commit a transaction, but is only used to commit a streaming transaction.

As far as I can see, the `_hotbackupCommitLock` lock is currently acquired by `transaction::Manager::holdTransactions()` in write mode. That will block the `transaction::Manager::commitManagedTrx()` from making progress and committing anything. Fine.
But as mentioned `commitManagedTrx` is used only to commit streaming transactions. It is not used to commit any other types of transactions, for example standalone AQL queries. 
So if I am not wrong, AQL queries will not be prevented from committing even if the write-lock on `_hotbackupCommitLock` is acquired by `holdTransactions()`.
This may or may not be intentional, just wanted to clarify it.

Reply:
Interesting, is this different on dbservers? This code was previously used on dbservers to stop all transactions. Maybe it is flawed as well there .

Reply:
I see. Is this different on dbservers? Is registerTransaction called for all transactions? Maybe it is already broken

Reply:
Should be fixed now.